web/src/lib/components/asset-viewer/photo-viewer.svelte

```diff
@@ -117,23 +195,32 @@
   }
 </script>
 
-<svelte:window on:keydown={handleKeypress} on:copyImage={doCopy} on:zoomImage={doZoomImage} />
+<svelte:window on:keydown={handleKeypress} on:rotateImage={doRotate} on:copyImage={doCopy} on:zoomImage={doZoomImage} />
 
 <div
   bind:this={element}
+  bind:clientHeight
+  bind:clientWidth
   transition:fade={{ duration: haveFadeTransition ? 150 : 0 }}
-  class="flex h-full select-none place-content-center place-items-center"
+  class="flex h-full w-full select-none place-content-center place-items-center"
 >
   {#await loadAssetData({ loadOriginal: false })}
     <LoadingSpinner />
   {:then}
-    <div bind:this={imgElement} class="h-full w-full">
+    <div
+      bind:this={imgElement}
+      class="duration-500"
+      style={asset.exifInfo?.orientation
+        ? `transform: rotate(${orientationToRotation(asset.exifInfo?.orientation)}deg);`
```

Comment:
Pretty sure this will conflict with thumbnails which already take rotation into account when they are generated.

Reply:
Should we trigger thumbnail generation when rotation changes, and then just always use the thumbnail as is?

Reply:
Yeah, currently all the webUI (with the correct width / height / animation) is done. Now, I'm trying to figuring out how to use correctly the orientation information. 

Another issue is live photos (google disables the live part when the image orientation is changed).